test/fixtures/dialects/bigquery/select_datetime.yml

```diff
@@ -26,15 +26,17 @@ file:
             function_name:
               function_name_identifier: EXTRACT
             bracketed:
-              start_bracket: (
-              date_part: DATE
-              keyword: FROM
-              expression:
+            - start_bracket: (
+            - expression:
+                column_reference:
+                  identifier: DATE
+            - keyword: FROM
+            - expression:
```

Comment:
@tunetheweb FYI I originally tried this same method but didn't submit because of this.

In this context in `EXTRACT` `DATE` is a vaild `date_part` (or maybe a `keyword`, don't know strictly which as the original approach may have been a bit of a hack) and should therefore be controlled by L010.
`EXTRACT(DATE FROM gmv.order_created_at)` converts the datetime value in the column into a `DATE`. So making this change would just introduce the opposite bug into `DATE` from `EXTRACT`.

Feels like the current `FunctionContentsGrammar` approach is too general to be applied to all functions. Interestingly in [ANSI](https://github.com/sqlfluff/sqlfluff/blob/80be738545d39e578c830b4347af3f44797d3833/src/sqlfluff/dialects/dialect_ansi.py#L3363-L3372) the `DATEADD` function seems to be separated out from the other functions but this is overwritten in bigquery.

Reply:
Actually this is not valid BigQuery syntax!

![image](https://user-images.githubusercontent.com/10931297/148964946-d135c762-717e-4c30-8872-aff3c43606c4.png)

Will update the date to do a real datepart.

Reply:
Cool thanks, let's see what the OP says.